src/instructlab/sdg/utils/chunkers.py

```diff
@@ -0,0 +1,609 @@
+            chunking_criteria (callable): Custom function for determining chunk breaks.
+        Returns:
+            list: List of document chunks.
+        """
+        current_buffer = []
+        document_chunks = []
+        prev_page_number = None
+        book_title = None
+
+        for idx, book_element in enumerate(json_book["main-text"]):
+            if book_element["type"] in [
+                "page-footer",
+                "picture",
+                "reference",
+                "meta-data",
+                "figure",
+                "page-header",
+            ]:
+                continue
+            if book_element["type"] == "footnote":
+                current_book_page_number = book_element["prov"][0]["page"]
+            elif book_element["type"] in [
+                "subtitle-level-1",
+                "paragraph",
+                "table",
+                "title",
+                "equation",
+            ]:  # 'page-header',
+                if book_element["type"] == "table":
+                    current_book_page_number = self.get_table_page_number(
+                        json_book, idx
+                    )
+                else:
+                    current_book_page_number = book_element["prov"][0]["page"]
+                    book_text = book_element["text"]
+
+                if book_element["type"] == "subtitle-level-1":
+                    if book_title is None:
+                        book_title = book_text
+                        book_text = f"# Title: **{book_text}**"
+                    else:
+                        book_text = f"## **{book_text}**"
+
+                if book_element["type"] == "title":
+                    book_text = f"# **{book_text}**"
+                if book_element["type"] == "page-header":
+                    book_text = f"Page Header: **{book_text}**\n\n"
+
+                if chunking_criteria is not None:
+                    # custom break function that can be used to chunk document
+                    if chunking_criteria(book_text):
+                        document_chunks.append("\n\n".join(current_buffer))
+                        current_buffer = []
+                elif (
+                    prev_page_number is not None
+                    and prev_page_number != current_book_page_number
+                ) and keep_same_page_thing_together:
+                    document_chunks.append("\n\n".join(current_buffer))
+                    current_buffer = []
+                else:
+                    if (
+                        self.get_token_count("\n\n".join(current_buffer), tokenizer)
+                        >= max_token_per_chunk
+                        and len(current_buffer) > 1
+                    ):
+                        chunk_text = "\n\n".join(current_buffer[:-1])
+                        print(
+                            f"Current chunk size {self.get_token_count(chunk_text, tokenizer)} and max is {max_token_per_chunk}"
```

Comment:
Is this print statement really helpful?  I would expect something like this in a debug level logging event or something like that.